FE#21: Added base dashboard.

logicplace opened the pull request:
Added basic dashboard with selectable control icons and a basic menu. Menu allows user to set their default view and logout (redirects to login page). Default view is loaded and selected on startup.

logicplace committed:
FE#21: Added base dashboard.

kbrimm commented on dashboard.html:
```diff
@@ -0,0 +1,74 @@
+        <script src="http://code.jquery.com/jquery-2.0.0.js"></script>
+        <link rel="stylesheet" href="css/dashboard.css"/>
+        <script type="text/javascript" src="js/dashboard.js"></script>
+        <script src="https://maps.googleapis.com/maps/api/js?callback=readyMap" async defer></script>
```
Will change to Leaflet - can do this as part of the Map API task.

kbrimm commented on dashboard.html:
```diff
@@ -0,0 +1,74 @@
+            </div>
+        </div>
+    </body>
+</html>
```
EOF newline

kbrimm commented on js/dashboard.js:
```diff
@@ -0,0 +1,136 @@
+        // TODO: log user out.
+        window.location = "login.html";
+    });
+});
```
EOF newline

kbrimm commented on css/dashboard.css:
```diff
@@ -0,0 +1,133 @@
+
+.centered {
+    text-align: center;
+}
```
EOF newline

kbrimm commented on css/dashboard.css:
```diff
@@ -0,0 +1,133 @@
+}
+
+.data-button-label > input:checked + .data-button {
+    background-color: lightgray;
```
This color change is a bit subtle - maybe change to something a little more obvious?

logicplace replied:
Suggestions?

kbrimm commented:
This looks great! I do have a couple of thoughts about improvement:
- I like the overall styling of the settings menu, but it feels a bit cramped
  - The whitespace around Logout is inconsistent with the rest of the menu. Adding some padding to that element would make the flow feel more unified.
  - The icons feel smooshed. Would it be possible to resize the images in the menu?
- Eventually we should think about unifying the theming between the login and dashboard pages, but that's a bridge we can cross later on.
- The menu/buttons don't display in IE 11 on Windows 7. I know we axed compatibility concerns with IE, it just makes me nervous and I thought I should point it out.

logicplace committed:
FE#21: newlines (thought that was auto wth)

logicplace committed:
FE#21: Fixed menu spacing and icon size.

logicplace committed:
FE#21: Added menu spacer to stretch it out a bit more. Probably temporary.

logicplace commented:
Unsurprisingly IE11 doesn't support var or calc. :( I'll remove the vars I guess. Also yeah this round wasn't so much about theming, more just placement & function. I think that will be a separate ticket.

logicplace committed:
FE#21: Removed CSS vars and calcs :'(

logicplace committed:
FE#21: Running under file:// schema on IE11, localStorage does not exist, so this prevents that error from being thrown.

logicplace committed:
FE#21: A more apparent 'checked' color for data view selection

logicplace commented:
I think I addressed everything. I tested on IE11 to make sure it works, too (other than saving the preference). I also added a temporary spacer to the menu so hopefully it looks a little less cramped overall?

kbrimm commented:
Looks good!
